File: minigames/src/main/java/dex/qixils/aviary/command/ArenaCreateCommand.java
package dex.qixils.aviary.command;

import dex.qixils.aviary.Minigames;
import dex.qixils.aviary.utils.AnvilPrompt;
import net.kyori.adventure.text.Component;
import net.minestom.server.command.builder.Command;
import net.minestom.server.entity.Player;

import java.util.concurrent.CompletableFuture;

public class ArenaCreateCommand extends Command {
	public ArenaCreateCommand() {
		super("create");
		setDefaultExecutor((sender, context) -> {
			if (!(sender instanceof Player player)) {
				Minigames.alert(sender, "This command can only be executed by players.");
				return;
			}
			CompletableFuture<String> nameFuture = AnvilPrompt.stringBuilder()
					.player(player)
					.prompt(Component.text("Please enter the ID of the arena to create"))
					.build();
			nameFuture.thenAccept(name -> {
				// TODO: Prompt for game type
				// TODO: Create arena
			});
		});
	}
}

File: minigames/src/main/kotlin/dex/qixils/aviary/minigames/command/ArenaCreateCommand.kt
package dex.qixils.aviary.minigames.command

import dex.qixils.aviary.minigames.Minigames
import dex.qixils.aviary.minigames.utils.AnvilPrompt
import net.kyori.adventure.text.Component
import net.minestom.server.command.builder.Command
import net.minestom.server.entity.Player

class ArenaCreateCommand : Command("create") {
	init {
		setDefaultExecutor { sender, _ ->
			if (sender !is Player) {
				Minigames.alert(sender, "This command can only be executed by players.")
				return@setDefaultExecutor
			}

			AnvilPrompt.stringBuilder()
				.player(sender)
				.prompt(Component.text("Please enter the ID of the arena to create"))
				.build()
				.thenAccept { name ->
					// TODO: Prompt for game type
					// TODO: Create arena
				}
		}
	}
}
